File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for(int c=0;c<commands.size();c++) {
        int i = commands[c][0];
        int j = commands[c][1];
        int k = commands[c][2];
        
        vector<int> tmp(array.begin()+i-1, array.begin()+j);
        sort(tmp.begin(), tmp.end());
        answer.push_back(tmp[k-1]);
    }
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.swift
import Foundation

func solution(_ array:[Int], _ commands:[[Int]]) -> [Int] {
    var answer = [Int]()
    for com in commands {
        let i = com[0], j = com[1], k = com[2]
        let subArr = Array(array[i-1..<j]).sorted()
        answer.append(subArr[k-1])
    }
    return answer
}